File: php/largest_prime_factor.php
<?php
function largest_prime_factor($x) {
    $largest = 2;
    $max = 0;
    while ((floor(sqrt($x)) + 1) > $largest) {
        while ($x % $largest == 0) {
            $x = $x / $largest;
            $max = $largest;
        }
        $largest += 1;
    }
    if ($x == 1) {
        return $max;
    } else {
        return $x;
    }
}

if (basename(__FILE__) === basename($_SERVER['PHP_SELF'])) {
    echo largest_prime_factor(600851475143);
}
?>

File: python/largest_prime_factor.py
import math

def largest_prime_factor(x):
    max = 0
    for i in range(2, int(math.sqrt(x)) + 1, 1):
        while x % i == 0:
            x /= i
            max = i
    if x == 1.0:
        return max
    else:
        return x

if __name__ == '__main__':
    print(largest_prime_factor(600851475143))